lib/ansible/module_utils/ovirt.py

```diff
@@ -0,0 +1,544 @@
+    :return: name of the entity, which link points to
+    """
+
+    if link:
+        return connection.follow_link(link).name
+    else:
+        return None
+
+
+def equal(param1, param2):
+    """
+    Compare two parameters and return if they are equal.
+    This parameter doesn't run equal operation if first parameter is None.
+    With this approach we don't run equal operation in case user don't
+    specify parameter in their task.
+
+    :param param1: user inputted parameter
+    :param param2: value of entity parameter
+    :return: True if parameters are equal or first parameter is None, otherwise False
+    """
+    if param1 is not None:
+        return param1 == param2
+    return True
+
+
+def search_by_attributes(service, **kwargs):
+    """
+    Search for the entity by attributes. Nested entities don't support search
+    via REST, so in case using search for nested entity we return all entities
+    and filter them by specified attributes.
+    """
+    # Check if 'list' method support search(look for search parameter):
+    if 'search' in inspect.getargspec(service.list)[0]:
+        res = service.list(
+            search=' and '.join('{}={}'.format(k, v) for k, v in kwargs.items())
+        )
+    else:
+        res = [
+            e for e in service.list() if len([
+                 k for k, v in kwargs.items() if getattr(e, k, None) == v
+            ]) == len(kwargs)
+        ]
+
+    res = res or [None]
+    return res[0]
+
+
+def search_by_name(service, name, **kwargs):
+    """
+    Search for the entity by its name. Nested entities don't support search
+    via REST, so in case using search for nested entity we return all entities
+    and filter them by name.
+
+    :param service: service of the entity
+    :param name: name of the entity
+    :return: Entity object returned by Python SDK
+    """
+    # Check if 'list' method support search(look for search parameter):
+    if 'search' in inspect.getargspec(service.list)[0]:
+        res = service.list(
+            search="name={name}".format(name=name)
+        )
+    else:
+        res = [e for e in service.list() if e.name == name]
+
+    if kwargs:
+        res = [
+            e for e in service.list() if len([
+                k for k, v in kwargs.items() if getattr(e, k, None) == v
+            ]) == len(kwargs)
+        ]
+
+    res = res or [None]
+    return res[0]
+
+
+def wait(service, condition, fail_condition=lambda e: False, timeout=180, wait=True):
+    """
+    Wait until entity fulfill expected condition.
+
+    :param service: service of the entity
+    :param condition: condition to be fulfilled
+    :param fail_condition: if this condition is true, raise Exception
+    :param timeout: max time to wait in seconds
+    :param wait: if True wait for condition, if False don't wait
+    """
+    # Wait until the desired state of the entity:
+    if wait:
+        start = time.time()
+        while time.time() < start + timeout:
```

Comment:
It'd be good to include a default sleep here so this loop doesn't continuously hit the oVirt endpoint. Even just a one or two second sleep would be fine between tries.